_inc/lib/core-api/class.jetpack-core-api-site-endpoints.php

```diff
@@ -69,6 +69,17 @@ public static function get_related_posts( $api_request ) {
 			);
 		}
 
+		// Activate the Related Posts module if it's not currently active
+		if ( ! Jetpack::is_module_active( 'related-posts' ) ) {
+			if ( ! Jetpack::activate_module( 'related-posts' ) ) {
```

Comment:
I'm not very familiar with module activations, but seems like if we don't want to `redirect` or `exit` on module activation, it would be safer to do this here:

```php
Jetpack::activate_module( 'related-posts', false, false )
```

See https://github.com/Automattic/jetpack/blob/f29016b02aa81b1021283062e1e0943d2599bb70/class.jetpack.php#L2899

Reply:
Good point 👍  I've done that in https://github.com/Automattic/jetpack/pull/10288/commits/7730aaa578ac392a8423c929d6a8e9eacf729cb4

Simply adding those was causing a fatal, though, because the related posts files module classes were not loaded when we activate a module after initialization. So we have to reload the modules after activating the module, which we're also doing in this commit.